packages/cfd/src/Containers/cfd-dashboard.tsx

```diff
@@ -301,18 +303,20 @@ const CFDDashboard = (props: TCFDDashboardProps) => {
     };
 
     const openAccountTransfer = (
-        data: DetailsOfEachMT5Loginid & { account_id?: string; platform?: string },
-        meta: { category: string; type?: string }
+        data: TNewDetailsOfEachMT5Loginid | TTradingPlatformAccounts,
+        meta: TOpenAccountTransferMeta
     ) => {
         if (meta.category === 'real') {
-            if (data.platform === CFD_PLATFORMS.DXTRADE)
-                sessionStorage.setItem('cfd_transfer_to_login_id', data.account_id as string);
-            else sessionStorage.setItem('cfd_transfer_to_login_id', data.login as string);
-
-            props.disableCFDPasswordModal();
-            props.history.push(routes.cashier_acc_transfer);
+            if (data) {
+                if ('platform' in data && data.platform === CFD_PLATFORMS.DXTRADE)
+                    sessionStorage.setItem('cfd_transfer_to_login_id', data.account_id as string);
+                else sessionStorage.setItem('cfd_transfer_to_login_id', data.login as string);
```

Comment:
```suggestion
                if ('platform' in data && data.platform === CFD_PLATFORMS.DXTRADE && typeof data.account_id === 'string')
                    sessionStorage.setItem('cfd_transfer_to_login_id', data.account_id as string);
                else if ( typeof data.login === 'string') sessionStorage.setItem('cfd_transfer_to_login_id', data.login as string);
```

Since there is a change for `data.account_id` or `data.login` to be null or undefined, it is better to check for it before setting it. You can do it like the above or have a default fallback to an empty string like below

```
if ('platform' in data && data.platform === CFD_PLATFORMS.DXTRADE)
                    sessionStorage.setItem('cfd_transfer_to_login_id', data.account_id || '');
                else sessionStorage.setItem('cfd_transfer_to_login_id', data.login || '');
```
Using the `as` assertion should always be your last resort: https://exploringjs.com/tackling-ts/ch_type-assertions.html

Reply:
Done